.github/workflows/test.yml

```diff
@@ -73,32 +73,19 @@ jobs:
     - name: Check out the repository
       uses: actions/checkout@v3.3.0
 
-    - name: Setup Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4.5.0
-      with:
-        python-version: ${{ matrix.python-version }}
-        architecture: x64
-        cache: 'pip'
-        cache-dependency-path: 'poetry.lock'
-
     - name: Install Poetry
       env:
         PIP_CONSTRAINT: .github/workflows/resources/constraints.txt
       run: |
         pipx install poetry
         poetry --version
 
-    - name: Upgrade pip
-      env:
-        PIP_CONSTRAINT: .github/workflows/resources/constraints.txt
-      run: |
-        pip install pip
-        pip --version
-
-    - name: Configure Poetry
-      run: |
-        poetry env use "${{ matrix.python-version }}"
-        poetry debug info
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4.5.0
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: x64
+        cache: 'poetry'
```

Comment:
IIRC the reason I used the `pip` cache is because this workflow doesn't really run `poetry install`, but instead `pip install` under the hood via Nox.